packages/providers/src/provider.test.ts

```diff
@@ -152,4 +152,14 @@ describe('Provider', () => {
 
     expect(minGasPrice).toBeDefined();
   });
+
+  it('can change the provider url of the curernt instance', async () => {
+    const provider = new Provider('http://127.0.0.1:4000/graphql');
+
+    provider.connect('http://127.0.0.1:8080/graphql');
+
+    expect(async () => {
+      await provider.getNodeInfo();
+    }).rejects.toThrow();
```

Comment:
Calling a second method _(`getNodeInfo`)_ to validate the first one _(`connect`)_ is not optimal. Only one unit should be considered per test, using mocks if necessary, and the assertion should focus on the `operations` property — the only thing the `connect` method changes.

> Also, try to always error-match using `.toThrow(/my-expected-error/)` to avoid _false-positives_ — when _any_ error can make a test pass, bugs can be silenced.